helper/chart_data.js

```diff
@@ -181,49 +191,59 @@ async function instanceMultiselectFunc(data) {
 
 //Function for entity Observation and observation report's response creation
 exports.entityReportChart = async function (data,entityId,entityName,reportType) {
-    var obj;
-    var multiSelectArray = [];
-    var textArray = [];
-    var radioArray = [];
-    var sliderArray = [];
-    var numberArray = [];
-    var dateArray = [];
-    var noOfSubmissions = [];
-    var matrixArray = [];
+    let obj;
+    let multiSelectArray = [];
+    let textArray = [];
+    let radioArray = [];
+    let sliderArray = [];
+    let numberArray = [];
+    let dateArray = [];
+    let noOfSubmissions = [];
+    let matrixArray = [];
     let actualData = data;
+    let type;
 
     try {
 
-        // obj is the response object which we are sending as a API response  
-        if (data[0].event.solutionId) {
-
+        if (reportType == filesHelper.survey) {
+            type = filesHelper.survey;
             obj = {
-                solutionId: data[0].event.solutionId,
                 solutionName: data[0].event.solutionName,
-                entityType: data[0].event.entityType,
-                entityId: entityId,
-                entityName: data[0].event[entityName + "Name"],
                 response: []
             }
-
-
         }
         else {
-            obj = {
-                observationId: data[0].event.observationId,
-                observationName: data[0].event.observationName,
-                entityType: data[0].event.entityType,
-                entityId: entityId,
-                entityName:  data[0].event[entityName + "Name"],
-                districtName: data[0].event.districtName,
-                response: []
+            type = filesHelper.observation;
+            if (data[0].event.solutionId) {
+
+                obj = {
+                    solutionId: data[0].event.solutionId,
+                    solutionName: data[0].event.solutionName,
+                    entityType: data[0].event.entityType,
```

Comment:
@deepakr123 - make a common object and then add keys based on if condition

Reply:
resolved